core/src/main/java/io/ultrabrew/metrics/data/ConcurrentMonoidLongTable.java

```diff
@@ -120,40 +118,19 @@
 
   /**
    * Create a simple linear probing hash table for a monoid operation.
-   *
-   * @param metricId identifier of the metric
-   * @param initialCapacity requested capacity of table in records, actual capacity may be higher
-   * @param fields sorted array of field names used in reporting
-   * @param types type of each corresponding field
-   * @param identity monoid's identity, where each value is corresponding to the given fields names
-   */
-  protected ConcurrentMonoidHashTable(final String metricId, final int initialCapacity,
-      final String[] fields, final Type[] types, final long[] identity) {
-    this(metricId, initialCapacity, fields, types, identity, DEFAULT_MAX_CAPACITY);
-  }
-
-  /**
-   * @param metricId identifier of the metric
+   *  @param metricId identifier of the metric
+   * @param maxCapacity maximum capacity of table in records.
    * @param initialCapacity requested capacity of table in records
    * @param fields sorted array of field names used in reporting
    * @param types type of each corresponding field
    * @param identity monoid's identity, where each value is corresponding to the given fields names
-   * @param maxCapacity maximum capacity of table in records.
    */
-  protected ConcurrentMonoidHashTable(final String metricId, int initialCapacity,
-      final String[] fields, final Type[] types, final long[] identity, final int maxCapacity) {
+  protected ConcurrentMonoidLongTable(final String metricId, final int maxCapacity,
+      int initialCapacity, final String[] fields, final Type[] types, final long[] identity) {
 
-    if (fields.length != identity.length || fields.length == 0) {
-      throw new IllegalArgumentException(
-          "Fields and Identity must match in length and be non-zero");
-    }
-    if (initialCapacity < 0) {
-      throw new IllegalArgumentException("Illegal initial capacity");
-    }
-    if (maxCapacity < initialCapacity) {
-      throw new IllegalArgumentException(
-          "max capacity should be greater than the initial capacity");
-    }
+    assert fields.length != 0 && fields.length == identity.length : "Fields and Identity must match in length and be non-zero";
```

Comment:
Please don't use `assert` in non-performance-critical code, almost no one ever runs JVM with assertions enabled.